### ISSUE TITLE
promotion of 2.03.05

### ISSUE THREAD

**Issue by ibaned** [kokkos/kokkos#828]
This issue tracks my progress on performing a Kokkos promotion.

**Comment by ibaned** [comment]
b4bc753362b323e43eafc1ec40fc41d219217279 will be used

**Comment by ibaned** [comment]
@crtrott what options are used for `test_all_sandia` ? I am on the `kokkos-dev` machine. Should I just run it with no arguments (do everything) first ?

**Comment by ibaned** [comment]
`test_all_sandia` passed on `kokkos-dev` with no arguments:
```
#######################################################
PASSED TESTS
#######################################################
clang-3.6.1-Pthread-hwloc-release build_time=121 run_time=104
clang-3.6.1-Pthread-release build_time=85 run_time=151
clang-3.6.1-Pthread_Serial-hwloc-release build_time=105 run_time=110
clang-3.6.1-Pthread_Serial-release build_time=95 run_time=165
clang-3.6.1-Serial-hwloc-release build_time=99 run_time=100
clang-3.6.1-Serial-release build_time=144 run_time=97
clang-3.7.1-Pthread-hwloc-release build_time=126 run_time=65
clang-3.7.1-Pthread-release build_time=125 run_time=159
clang-3.7.1-Pthread_Serial-hwloc-release build_time=103 run_time=131
clang-3.7.1-Pthread_Serial-release build_time=137 run_time=192
clang-3.7.1-Serial-hwloc-release build_time=111 run_time=106
clang-3.7.1-Serial-release build_time=90 run_time=129
clang-3.8.1-Pthread-hwloc-release build_time=124 run_time=61
clang-3.8.1-Pthread-release build_time=175 run_time=129
clang-3.8.1-Pthread_Serial-hwloc-release build_time=135 run_time=118
clang-3.8.1-Pthread_Serial-release build_time=158 run_time=186
clang-3.8.1-Serial-hwloc-release build_time=134 run_time=104
clang-3.8.1-Serial-release build_time=105 run_time=123
clang-3.9.0-Pthread-hwloc-release build_time=128 run_time=70
clang-3.9.0-Pthread-release build_time=163 run_time=139
clang-3.9.0-Pthread_Serial-hwloc-release build_time=134 run_time=91
clang-3.9.0-Pthread_Serial-release build_time=149 run_time=160
clang-3.9.0-Serial-hwloc-release build_time=147 run_time=95
clang-3.9.0-Serial-release build_time=116 run_time=121
cuda-7.0.28-Cuda_OpenMP-release build_time=409 run_time=513
cuda-7.0.28-Cuda_Pthread-release build_time=404 run_time=519
cuda-7.0.28-Cuda_Serial-release build_time=379 run_time=453
cuda-7.5.18-Cuda_OpenMP-release build_time=386 run_time=468
cuda-7.5.18-Cuda_Pthread-release build_time=381 run_time=466
cuda-7.5.18-Cuda_Serial-release build_time=369 run_time=448
cuda-8.0.44-Cuda_OpenMP-release build_time=350 run_time=457
cuda-8.0.44-Cuda_Pthread-release build_time=349 run_time=473
cuda-8.0.44-Cuda_Serial-release build_time=334 run_time=504
gcc-4.7.2-OpenMP-hwloc-release build_time=144 run_time=68
gcc-4.7.2-OpenMP-release build_time=142 run_time=101
gcc-4.7.2-OpenMP_Serial-hwloc-release build_time=134 run_time=152
gcc-4.7.2-OpenMP_Serial-release build_time=151 run_time=117
gcc-4.7.2-Pthread-hwloc-release build_time=114 run_time=56
gcc-4.7.2-Pthread-release build_time=140 run_time=109
gcc-4.7.2-Pthread_Serial-hwloc-release build_time=155 run_time=130
gcc-4.7.2-Pthread_Serial-release build_time=133 run_time=159
gcc-4.7.2-Serial-hwloc-release build_time=100 run_time=95
gcc-4.7.2-Serial-release build_time=90 run_time=97
gcc-4.8.4-OpenMP-hwloc-release build_time=108 run_time=121
gcc-4.8.4-OpenMP-release build_time=118 run_time=144
gcc-4.8.4-OpenMP_Serial-hwloc-release build_time=132 run_time=119
gcc-4.8.4-OpenMP_Serial-release build_time=126 run_time=165
gcc-4.8.4-Pthread-hwloc-release build_time=101 run_time=75
gcc-4.8.4-Pthread-release build_time=120 run_time=138
gcc-4.8.4-Pthread_Serial-hwloc-release build_time=119 run_time=148
gcc-4.8.4-Pthread_Serial-release build_time=139 run_time=179
gcc-4.8.4-Serial-hwloc-release build_time=111 run_time=94
gcc-4.8.4-Serial-release build_time=82 run_time=134
intel-14.0.4-OpenMP-hwloc-release build_time=331 run_time=254
intel-14.0.4-OpenMP-release build_time=413 run_time=237
intel-14.0.4-OpenMP_Serial-hwloc-release build_time=318 run_time=245
intel-14.0.4-OpenMP_Serial-release build_time=322 run_time=223
intel-14.0.4-Pthread-hwloc-release build_time=325 run_time=165
intel-14.0.4-Pthread-release build_time=319 run_time=313
intel-14.0.4-Pthread_Serial-hwloc-release build_time=315 run_time=198
intel-14.0.4-Pthread_Serial-release build_time=281 run_time=262
intel-14.0.4-Serial-hwloc-release build_time=251 run_time=209
intel-14.0.4-Serial-release build_time=247 run_time=200
intel-15.0.2-OpenMP-hwloc-release build_time=310 run_time=212
intel-15.0.2-OpenMP-release build_time=310 run_time=228
intel-15.0.2-OpenMP_Serial-hwloc-release build_time=328 run_time=225
intel-15.0.2-OpenMP_Serial-release build_time=330 run_time=260
intel-15.0.2-Pthread-hwloc-release build_time=278 run_time=193
intel-15.0.2-Pthread-release build_time=299 run_time=201
intel-15.0.2-Pthread_Serial-hwloc-release build_time=309 run_time=257
intel-15.0.2-Pthread_Serial-release build_time=339 run_time=220
intel-15.0.2-Serial-hwloc-release build_time=286 run_time=179
intel-15.0.2-Serial-release build_time=229 run_time=214
intel-16.0.1-OpenMP-hwloc-release build_time=410 run_time=169
intel-16.0.1-OpenMP-release build_time=397 run_time=215
intel-16.0.1-OpenMP_Serial-hwloc-release build_time=401 run_time=211
intel-16.0.1-OpenMP_Serial-release build_time=398 run_time=239
intel-16.0.1-Pthread-hwloc-release build_time=288 run_time=209
intel-16.0.1-Pthread-release build_time=286 run_time=245
intel-16.0.1-Pthread_Serial-hwloc-release build_time=320 run_time=303
intel-16.0.1-Pthread_Serial-release build_time=319 run_time=234
intel-16.0.1-Serial-hwloc-release build_time=259 run_time=202
intel-16.0.1-Serial-release build_time=297 run_time=177
```
Step 1.3 complete

**Comment by ibaned** [comment]
Step 1 complete

**Comment by ibaned** [comment]
Step 2 complete on `kokkos-dev`

Step 3 in progress on White. Don't have Shepard account, WebCARS request submitted.

**Comment by hcedwar** [comment]
Don't let waiting for shepard account hold things up - someone else on the team can do the testing there.

**Comment by ndellingwood** [comment]
All shepard tests pass:

Tested develop branch sha b4bc753362b323e43eafc1ec40fc41d219217279

```
#######################################################
PASSED TESTS
#######################################################
intel-16.2.181-Pthread-release build_time=281 run_time=135
intel-16.2.181-Pthread_Serial-release build_time=292 run_time=177
intel-16.2.181-Serial-release build_time=236 run_time=121
intel-17.0.098-OpenMP-release build_time=337 run_time=123
intel-17.0.098-OpenMP_Serial-release build_time=351 run_time=164
intel-17.0.098-Pthread-release build_time=293 run_time=132
intel-17.0.098-Pthread_Serial-release build_time=323 run_time=169
intel-17.0.098-Serial-release build_time=241 run_time=118
#######################################################
```

**Comment by ibaned** [comment]
Should be integration testing with the two scripts here:

https://github.com/kokkos/kokkos/tree/develop/config/trilinos-integration

**Comment by ibaned** [comment]
`shepard` account approved. will launch integration tests there.

**Comment by ibaned** [comment]
not quite. only approved by manager, waiting on `wg-advanced-arch`.

**Comment by ibaned** [comment]
Configuration failures in `white` Trilinos integration testing:
```
  Cannot find source file:

    TestDefaultDeviceTypeInit_6.cpp
--
  Cannot find source file:

    TestDefaultDeviceTypeInit_5.cpp
--
  Cannot find source file:

    TestDefaultDeviceTypeInit_4.cpp
--
  Cannot find source file:

    TestDefaultDeviceTypeInit_2.cpp
--
  Cannot find source file:

    TestDefaultDeviceTypeInit_1.cpp
--
  Cannot find source file:

    TestDefaultDeviceTypeInit_12.cpp
--
  Cannot find source file:

    TestDefaultDeviceTypeInit_3.cpp
--
  Cannot find source file:

    TestDefaultDeviceTypeInit_7.cpp
--
  Cannot find source file:

    TestDefaultDeviceTypeInit_8.cpp
--
  Cannot find source file:

    TestDefaultDeviceTypeInit_9.cpp
--
  Cannot find source file:

    TestDefaultDeviceTypeInit_10.cpp
--
  Cannot find source file:

    serial/TestSerial_ViewAPI_a.cpp
--
  Cannot find source file:

    TestDefaultDeviceTypeInit_15.cpp
--
  Cannot find source file:

    cuda/TestCuda_Reductions_a.cpp
--
  Cannot find source file:

    TestDefaultDeviceTypeInit_11.cpp
--
  Cannot find source file:

    TestDefaultDeviceTypeInit_13.cpp
--
  Cannot find source file:

    TestDefaultDeviceType.cpp
--
  Cannot find source file:

    TestDefaultDeviceTypeInit_14.cpp
--
  Cannot find source file:

    TestDefaultDeviceTypeInit_16.cpp
--
  Cannot find source file:

    PerfTestHost.cpp
```
I see they were moved, working on resolution.

**Comment by ibaned** [comment]
@hcedwar @crtrott I could use write access to Kokkos develop and master branches.

**Comment by ibaned** [comment]
Have commit ready to fix config failures. Assuming direct write access is best, although could create PR if requested.

**Comment by ibaned** [comment]
opened PR #830 which should fix TriBITS. Issue #832 documents failing tests with Trilinos, with both Serial and OpenMP enabled.

**Comment by ibaned** [comment]
Several merges just happened: #830, #833, #827, #823. I created #836, which is not merged yet, but with that one included, in theory, all known blocking issues should be fixed. Restarting from Step 1, running `test_all_sandia` for #836.

**Comment by ibaned** [comment]
#836 merged, launched Trilinos integration testing on `white`. @crtrott will launch on `shepard`.

**Comment by ibaned** [comment]
`white_cuda` build error:
```
/ascldap/users/daibane/kokkos-promotion/white_cuda/trilinos-update/packages/kokkos/core/unit_test/cuda/TestCuda_Spaces.cpp(51): error: name followed by "::" must be a class or namespace name
```

**Comment by ibaned** [comment]
`white_omp` build error:
```
/ascldap/users/daibane/kokkos-promotion/white_omp/trilinos-update/packages/kokkos-kernels/src/impl/Kokkos_Sparse_impl_spmv_omp.hpp:63:29: error: missing expression between '(' and ')'
 #if (KOKKOS_ENABLE_PROFILING)
                             ^
```

**Comment by ibaned** [comment]
PR #837 should fix `white_cuda` build

**Comment by ibaned** [comment]
#837 accepted after testing on Pascal. Several build errors introduced in Trilinos, fixed in branches of Trilinos repo. These commits will be prepended to the kokkos snapshot commit to Trilinos.

SEMS people broke our modules this morning. They have now fixed them again. `test_all_sandia` started again on `kokkos-dev`, passing so far.

Modifying Trilinos integration scripts to accept Trilinos branch...

**Comment by ibaned** [comment]
In #839, scripts accept branch arguments. Using this to do Trilinos integration testing with extra fix commits added to Trilinos branch. Started on `white` like so:
```bash
nohup ../kokkos/config/trilinos-integration/white_run_jenkins_script_omp integration-script-branch kokkos-promotion > dan_log.txt &
```

**Comment by ibaned** [comment]
`test_all_sandia` passed on `kokkos-dev` for ad79c0c:
```
#######################################################
PASSED TESTS
#######################################################
clang-3.6.1-Pthread-hwloc-release build_time=114 run_time=92
clang-3.6.1-Pthread-release build_time=88 run_time=152
clang-3.6.1-Pthread_Serial-hwloc-release build_time=107 run_time=110
clang-3.6.1-Pthread_Serial-release build_time=96 run_time=170
clang-3.6.1-Serial-hwloc-release build_time=96 run_time=100
clang-3.6.1-Serial-release build_time=129 run_time=90
clang-3.7.1-Pthread-hwloc-release build_time=121 run_time=90
clang-3.7.1-Pthread-release build_time=106 run_time=155
clang-3.7.1-Pthread_Serial-hwloc-release build_time=114 run_time=98
clang-3.7.1-Pthread_Serial-release build_time=120 run_time=177
clang-3.7.1-Serial-hwloc-release build_time=122 run_time=102
clang-3.7.1-Serial-release build_time=106 run_time=123
clang-3.8.1-Pthread-hwloc-release build_time=143 run_time=73
clang-3.8.1-Pthread-release build_time=142 run_time=121
clang-3.8.1-Pthread_Serial-hwloc-release build_time=134 run_time=127
clang-3.8.1-Pthread_Serial-release build_time=149 run_time=188
clang-3.8.1-Serial-hwloc-release build_time=179 run_time=87
clang-3.8.1-Serial-release build_time=117 run_time=138
clang-3.9.0-Pthread-hwloc-release build_time=124 run_time=97
clang-3.9.0-Pthread-release build_time=178 run_time=133
clang-3.9.0-Pthread_Serial-hwloc-release build_time=132 run_time=92
clang-3.9.0-Pthread_Serial-release build_time=157 run_time=160
clang-3.9.0-Serial-hwloc-release build_time=165 run_time=101
clang-3.9.0-Serial-release build_time=105 run_time=142
cuda-7.0.28-Cuda_OpenMP-release build_time=498 run_time=619
cuda-7.0.28-Cuda_Pthread-release build_time=818 run_time=618
cuda-7.0.28-Cuda_Serial-release build_time=507 run_time=533
cuda-7.5.18-Cuda_OpenMP-release build_time=462 run_time=583
cuda-7.5.18-Cuda_Pthread-release build_time=498 run_time=556
cuda-7.5.18-Cuda_Serial-release build_time=485 run_time=525
cuda-8.0.44-Cuda_OpenMP-release build_time=416 run_time=573
cuda-8.0.44-Cuda_Pthread-release build_time=399 run_time=567
cuda-8.0.44-Cuda_Serial-release build_time=516 run_time=525
gcc-4.7.2-OpenMP-hwloc-release build_time=149 run_time=78
gcc-4.7.2-OpenMP-release build_time=151 run_time=104
gcc-4.7.2-OpenMP_Serial-hwloc-release build_time=143 run_time=130
gcc-4.7.2-OpenMP_Serial-release build_time=155 run_time=119
gcc-4.7.2-Pthread-hwloc-release build_time=121 run_time=56
gcc-4.7.2-Pthread-release build_time=148 run_time=107
gcc-4.7.2-Pthread_Serial-hwloc-release build_time=140 run_time=684
gcc-4.7.2-Pthread_Serial-release build_time=143 run_time=160
gcc-4.7.2-Serial-hwloc-release build_time=106 run_time=96
gcc-4.7.2-Serial-release build_time=104 run_time=97
gcc-4.8.4-OpenMP-hwloc-release build_time=114 run_time=685
gcc-4.8.4-OpenMP-release build_time=116 run_time=656
gcc-4.8.4-OpenMP_Serial-hwloc-release build_time=131 run_time=130
gcc-4.8.4-OpenMP_Serial-release build_time=129 run_time=171
gcc-4.8.4-Pthread-hwloc-release build_time=134 run_time=86
gcc-4.8.4-Pthread-release build_time=134 run_time=144
gcc-4.8.4-Pthread_Serial-hwloc-release build_time=122 run_time=106
gcc-4.8.4-Pthread_Serial-release build_time=143 run_time=162
gcc-4.8.4-Serial-hwloc-release build_time=112 run_time=94
gcc-4.8.4-Serial-release build_time=78 run_time=145
intel-14.0.4-OpenMP-hwloc-release build_time=291 run_time=187
intel-14.0.4-OpenMP-release build_time=321 run_time=186
intel-14.0.4-OpenMP_Serial-hwloc-release build_time=319 run_time=240
intel-14.0.4-OpenMP_Serial-release build_time=328 run_time=229
intel-14.0.4-Pthread-hwloc-release build_time=294 run_time=166
intel-14.0.4-Pthread-release build_time=265 run_time=209
intel-14.0.4-Pthread_Serial-hwloc-release build_time=309 run_time=196
intel-14.0.4-Pthread_Serial-release build_time=286 run_time=270
intel-14.0.4-Serial-hwloc-release build_time=249 run_time=206
intel-14.0.4-Serial-release build_time=243 run_time=200
intel-15.0.2-OpenMP-hwloc-release build_time=305 run_time=259
intel-15.0.2-OpenMP-release build_time=306 run_time=234
intel-15.0.2-OpenMP_Serial-hwloc-release build_time=338 run_time=227
intel-15.0.2-OpenMP_Serial-release build_time=332 run_time=272
intel-15.0.2-Pthread-hwloc-release build_time=288 run_time=176
intel-15.0.2-Pthread-release build_time=320 run_time=198
intel-15.0.2-Pthread_Serial-hwloc-release build_time=304 run_time=232
intel-15.0.2-Pthread_Serial-release build_time=344 run_time=221
intel-15.0.2-Serial-hwloc-release build_time=277 run_time=178
intel-15.0.2-Serial-release build_time=229 run_time=192
intel-16.0.1-OpenMP-hwloc-release build_time=395 run_time=172
intel-16.0.1-OpenMP-release build_time=404 run_time=203
intel-16.0.1-OpenMP_Serial-hwloc-release build_time=385 run_time=231
intel-16.0.1-OpenMP_Serial-release build_time=378 run_time=231
intel-16.0.1-Pthread-hwloc-release build_time=292 run_time=224
intel-16.0.1-Pthread-release build_time=289 run_time=229
intel-16.0.1-Pthread_Serial-hwloc-release build_time=323 run_time=276
intel-16.0.1-Pthread_Serial-release build_time=312 run_time=250
intel-16.0.1-Serial-hwloc-release build_time=268 run_time=195
intel-16.0.1-Serial-release build_time=295 run_time=184
```

**Comment by ibaned** [comment]
`white` integration tests were missing #837... restarting them.

**Comment by ibaned** [comment]
`white` OpenMP integration test looks good, for 481a398

**Comment by ibaned** [comment]
`shepard` account approved, launched both integration scripts there

**Comment by ibaned** [comment]
All integration tests are run with trilinos/Trilinos@2f3c8c6

**Comment by ibaned** [comment]
`white` CUDA integration compile failure:
```
/ascldap/users/daibane/kokkos-promotion/white_cuda/trilinos-update/packages/sacado/src/Sacado_DynamicArrayTraits.hpp(75): error: namespace "Kokkos::Experimental" has no member "MemoryPool"
```

**Comment by ibaned** [comment]
`shepard` integrations were missing
```
git config --global user.*
```
restarting them.

**Comment by ibaned** [comment]
trilinos/Trilinos@ba21864480 addresses the Sacado build failure on CUDA

**Comment by ibaned** [comment]
`white` CUDA integration tests restarted

**Comment by ibaned** [comment]
nevermind, lots of Sacado tests need updating

**Comment by ibaned** [comment]
trilinos/Trilinos@2c04eb8fa0 addresses the Sacado unit tests as well

**Comment by ibaned** [comment]
Linker error on `shepard`, seems unrelated to our changes:
```
ld: /home/projects/x86-64-haswell/boost/1.59.0/openmpi/1.10.0/intel/16.1.056/lib/libboost_program_options.a(cmdline.o): relocation R_X86_64_32 against `_ZNSs4_Rep20_S_empty_rep_storageE' can not be used when making a shared object; recompile with -fPIC
/home/projects/x86-64-haswell/boost/1.59.0/openmpi/1.10.0/intel/16.1.056/lib/libboost_program_options.a: error adding symbols: Bad value
make[2]: *** [packages/stk/stk_util/stk_util/environment/libstk_util_env.so.12.11] Error 1
```

**Comment by ibaned** [comment]
Switched to static libraries on `shepard`, now I get an internal compiler error:
```
[ 96%] Building CXX object packages/stokhos/example/CMakeFiles/Stokhos_cijk_partition_zoltan.dir/cijk/cijk_partition_zoltan.cpp.o
": internal error: ** The compiler has encountered an unexpected problem.
** Segmentation violation signal raised. **
Access violation or stack overflow. Please contact Intel Support for assistance.

icpc: error #10014: problem during multi-file optimization compilation (code 4)
```

**Comment by ibaned** [comment]
- `white` CUDA build compiled and ran the "after" copy, results look tentatively good. It is currently compiling the "before" Trilinos copy.

 - `white` OpenMP build seems to be good (!), same test output before and after Kokkos snapshot.

 - `shepard` Threads build had the following error with new Kokkos:
```
/home/daibane/pthread/trilinos-update/packages/teuchos/core/src/Teuchos_RCP.hpp(288): error: a value of type "const Tpetra::Map<int, int, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Threads, Kokkos::Threads::memory_space>> *" cannot be used to initialize an entity of type "const Tpetra::Map<int, int, panzer::TpetraNodeType> *"
    : ptr_(r_ptr.get()), // will not compile if T is not base class of T2
           ^
          detected during:
            instantiation of "Teuchos::RCP<T>::RCP(const Teuchos::RCP<T2> &) [with T=const Tpetra::Map<int, int, panzer::TpetraNodeType>, T2=const Tpetra::Map<int, int, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Threads, Kokkos::Threads::memory_space>>]" at line 116 of "/home/daibane/pthread/trilinos-update/packages/panzer/dof-mgr/src/Panzer_Filtered_UniqueGlobalIndexer_impl.hpp"
            instantiation of "void panzer::Filtered_UniqueGlobalIndexer<LocalOrdinalT, GlobalOrdinalT>::getOwnedAndGhostedNotFilteredIndicator(std::vector<int, std::allocator<int>> &) const [with LocalOrdinalT=int, GlobalOrdinalT=int]" at line 48 of "/home/daibane/pthread/trilinos-update/packages/panzer/dof-mgr/src/Panzer_Filtered_UniqueGlobalIndexer.cpp"

/home/daibane/pthread/trilinos-update/packages/teuchos/core/src/Teuchos_RCP.hpp(288): error: a value of type "const Tpetra::Map<int, panzer::Ordinal64={long long}, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Threads, Kokkos::Threads::memory_space>> *" cannot be used to initialize an entity of type "const Tpetra::Map<int, panzer::Ordinal64={long long}, panzer::TpetraNodeType> *"
    : ptr_(r_ptr.get()), // will not compile if T is not base class of T2
           ^
          detected during:
            instantiation of "Teuchos::RCP<T>::RCP(const Teuchos::RCP<T2> &) [with T=const Tpetra::Map<int, panzer::Ordinal64={long long}, panzer::TpetraNodeType>, T2=const Tpetra::Map<int, panzer::Ordinal64={long long}, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Threads, Kokkos::Threads::memory_space>>]" at line 116 of "/home/daibane/pthread/trilinos-update/packages/panzer/dof-mgr/src/Panzer_Filtered_UniqueGlobalIndexer_impl.hpp"
            instantiation of "void panzer::Filtered_UniqueGlobalIndexer<LocalOrdinalT, GlobalOrdinalT>::getOwnedAndGhostedNotFilteredIndicator(std::vector<int, std::allocator<int>> &) const [with LocalOrdinalT=int, GlobalOrdinalT=panzer::Ordinal64={long long}]" at line 51 of "/home/daibane/pthread/trilinos-update/packages/panzer/dof-mgr/src/Panzer_Filtered_UniqueGlobalIndexer.cpp"

compilation aborted for /home/daibane/pthread/trilinos-update/packages/panzer/dof-mgr/src/Panzer_Filtered_UniqueGlobalIndexer.cpp
```
and is currently compiling the "before" copy.

 - `shepard` Serial had the aforementioned internal compiler error (likely due to large static binaries), but  only a few tests failed. It is currently configuring the "before" copy.

**Comment by ibaned** [comment]
`white` CUDA build is done, and looks pretty good:
 - `Amesos2_Tpetra_CrsMatrix_Adapter_UnitTests_MPI_4` passed on `updated` and timed out in `pristine`. Probably has intermittent behavior.
 - `MueLu_Stratimikos2_MPI_4` and `PanzerAdaptersSTK_assembly_engine_MPI_4` timed out in `updated` and failed in `pristine`

`shepard` Threads build completed, also looks good. It had the same compile error in `pristine`, so it was not caused by Kokkos. Test output was the same except for `Belos_pseudo_pcg_hb_1_MPI_4` which passed on `updated` and was failing on `pristine`.

`shepard` Serial build completed, and also looks good (!). Same internal compiler errors on `pristine`, only a few things affected I think. Only test output difference is `Anasazi_Epetra_LOBPCG_solvertest_MPI_4` passed on `updated` and failed on `pristine`.

**Comment by ibaned** [comment]
Branches used were Kokkos `stage` and Trilinos `kokkos-promotion` for `updated`, Kokkos `master` and Trilinos `develop` for `pristine`. Kokkos `develop` should be equivalent to `stage`, same PRs were merged in both. All non-merge commits in Trilinos `kokkos-promotion` that are not in Trilinos `develop` need to accompany the Kokkos snapshot.

**Comment by crtrott** [comment]
I am running check-in-test now.

**Comment by crtrott** [comment]
Our configure scripts didn't explicitly add KokkosKernels. We need to fix that for next time. There was another sqrt namespace issue but I am fixing that explicitly and am going on. No need to rerun the tests. Was only in the fenl example.

**Comment by crtrott** [comment]
There is also some issues with some TPLS not correctly being found from the SEMS modules and some link issues in STK. I explicitly gave the TPL libraries and disabled the offending STK packages:
```
cat COMMON.config
-Dyaml-cpp_INCLUDE_DIRS=/projects/sems/install/rhel6-x86_64/sems/tpl/yaml_cpp/0.5.3/gcc/4.9.3/base/include
-Dyaml-cpp_LIBRARY_DIRS=/projects/sems/install/rhel6-x86_64/sems/tpl/yaml_cpp/0.5.3/gcc/4.9.3/base/lib
-DSuperLU_INCLUDE_DIRS=/projects/sems/install/rhel6-x86_64/sems/tpl/superlu/4.3/gcc/4.9.3/base/include
-DSuperLU_LIBRARY_DIRS=/projects/sems/install/rhel6-x86_64/sems/tpl/superlu/4.3/gcc/4.9.3/base/lib

nohup python ~/Trilinos/checkin-test.py --do-all --make-options="-j24" --disable-packages=STKDoc_tests,STKUtil,STKUnit_test_utils --push&
```

**Comment by crtrott** [comment]
Two failing tests at checkin:
```
1237 - Anasazi_Epetra_OrthoManagerGenTester_0_MPI_4 (Failed)
1657 - Phalanx_mdfield_runtime_MPI_1 (Failed)
```
The first one was caught by our testing and is intermittent. The second seems a real issue. For our Integration testing Phalanx was not enabled (another thing to add). This has something to do with bounds checking on DynRank View. I believe it is most likely an error in the phalanx test itself.

**Comment by crtrott** [comment]
Actually when running the phalanx test on develop it prints out an internal error, but reports the test as failed ...

**Comment by crtrott** [comment]
Nope it probably is a true issue ... Something to do with Phalanx using the [] operator to loop over the whole array, and bounds-checking on will prevent that for 2d or higher. Needs a fix ...

**Comment by crtrott** [comment]
I may look at it tonight again, but this puts the promotion on hold I guess ..

**Comment by ibaned** [comment]
I'll see if I can replicate it...

**Comment by crtrott** [comment]
I fixed it by getting rid of the bounds-check in the operator[]. I am running a spot-check to push it into develop and then run checkin-test again. Since I only commented out the bounds check I judge this to be extremely low impact probability so we don't need to rerun the white and shepard integration testing. That said the checkin-test will probably take again another 5-6 hours ...

**Comment by ibaned** [comment]
running just that one test with your fix seems to have the same behavior as `pristine`, i.e. still prints something about `any_cast` but the test passes. Hopefully checkin-test agrees.

**Comment by crtrott** [comment]
yeah thats what I am guessing. I did a spot check on develop which went ok, and now am rerunning the checkin-test

**Comment by crtrott** [comment]
almost done building, then its just another 2 hours for the tests to run (>2200 are enabled)

**Comment by crtrott** [comment]
Success :-)

**Comment by crtrott** [comment]
Closing this now.
